tests/logictest/suites/base/11_data_type/11_0005_data_type_date

```diff
@@ -0,0 +1,83 @@
+statement ok
+set timezone = 'UTC';
+
+statement query B
+SELECT '2022-02-03T00:00:00+09:00'::DATE = '2022-02-02'::DATE;
+
+----
+1
+
+statement query T
+select '0000-00-00'::DATE;
+
+----
+1970-01-01
+
+statement query T
+select '0099-05-16'::DATE;
+
+----
+1000-01-01
+
+statement query T
+select '1022-05-16'::DATE;
+
+----
+1022-05-16
+
+statement query T
+select '0000-00-00 00:00:00.868894'::DATE;
+
+----
+1970-01-01
+
+statement query T
+select '0099-05-16T03:25:02.868894'::DATE;
+
+----
+1000-01-01
+
+statement query T
+select '1022-05-16 03:25:02.868894'::DATE;
+
+----
+1022-05-16
+
+statement ok
+drop database if exists db1;
+
+statement ok
+create database db1;
+
+statement ok
+create table db1.t1(id int, a date);
+
+statement ok
+insert into db1.t1 values(1, '2022-01-01 02:00:11'), (2, '2022-01-02T02:00:22'), (3, '2022-02-02T04:00:03+00:00'), (4, '2022-03-02T01:12:00+07:00')
+
+statement query IT
+select * from db1.t1 order by id;
+
+----
+1  2022-01-01
+2  2022-01-02
+3  2022-02-02
+4  2022-03-01
+
+statement query I
+select id from db1.t1 where a = '2022-03-02T01:12:00+07:00';
```

Comment:
Since hours, minutes, and seconds are lost when the time string is converted to date, we should convert them to timestamp when comparing date with string, otherwise, it will lead to wrong results.